engines/supernova/state.h

```diff
@@ -0,0 +1,234 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef STATE_H
+#define STATE_H
+
+#include "common/rect.h"
+#include "common/keyboard.h"
+#include "supernova/rooms.h"
+
+namespace Supernova {
+
+const int32 INT32_MAX = 0x7FFFFFFF;
```

Comment:
This does not compile on my system using GCC 7.2.1 because `INT32_MAX` is already defined:
```
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/7.2.1/include/stdint.h:9:0,
                 from /usr/include/inttypes.h:27,
                 from /usr/include/SDL2/SDL_stdinc.h:69,
                 from /usr/include/SDL2/SDL_endian.h:31,
                 from /scummvm/common/scummsys.h:309,
                 from /scummvm/common/func.h:26,
                 from /scummvm/gui/debugger.h:26,
                 from /scummvm/engines/supernova/console.cpp:23:
/scummvm/engines/supernova/state.h:32:13: error: expected unqualified-id before numeric constant
 const int32 INT32_MAX = 0x7FFFFFFF;
```